cpp/src/gandiva/engine.cc

```diff
@@ -211,6 +213,7 @@ Status UseJITLinkIfEnabled(llvm::orc::LLJITBuilder& jit_builder) {
 
 Result<std::unique_ptr<llvm::orc::LLJIT>> BuildJIT(
     llvm::orc::JITTargetMachineBuilder jtmb,
+    std::unique_ptr<llvm::TargetMachine> target_machine,
```

Comment:
This is a minor change, after profiling, I found constructing `TargetMachine` takes some time, and here I tries to re-use the `TargetMachine` instance created previously.